packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -94,6 +96,44 @@ exports.sourceNodes = async (
     pluginConfig,
   })
 
+  // compare data between currentSyncData and previousSyncData and update it
+  previousSyncData.entries.filter(entry =>
+    _.findIndex(currentSyncData.deletedEntries, o => o.sys.id === entry.sys.id)
+  )
+
+  previousSyncData.assets.filter(asset =>
+    _.findIndex(currentSyncData.deletedAssets, o => o.sys.id === asset.sys.id)
+  )
+
+  // order is important here, fresh data first
+  currentSyncData.entries = _.uniqBy(
+    currentSyncData.entries.concat(previousSyncData.entries),
+    `sys.id`
+  )
+
+  currentSyncData.assets = _.uniqBy(
+    currentSyncData.assets.concat(previousSyncData.assets),
+    `sys.id`
+  )
+  // TODO run link resolution on the up to date data
+
+  /*
+   * on sync, we only get the synced data because there isn't enough data
+   * we want to trick the SDK to merge the data
+   * on initial sync, we save all the data
+   * on subsequent syncs, we get the updated data; merge with the old data
+   * if there are duplicate entries we take newest
+   */
+  const res = client.parseEntries({
+    items: currentSyncData.entries,
```

Comment:
According to the documentation, `parseEntries()` is expecting raw JSON. I believe what we are passing is not a plain object?

Reply:
By raw json the documentation means raw api output. That is what we are actually creating here manually.

This should be fine :)